pandas/io/pytables.py

```diff
@@ -3434,8 +3451,10 @@ def read_axes(self, where, **kwargs):
         # convert the data
         for a in self.axes:
             a.set_info(self.info)
+            # `kwargs` may contain `start` and `stop` arguments if passed to
+            # `store.select()`. If set they determine the index size.
             a.convert(values, nan_rep=self.nan_rep, encoding=self.encoding,
-                      errors=self.errors)
+                      errors=self.errors, **kwargs)
```

Comment:
where is this kwargs coming from? do you mean to pass ing start/stop?

Reply:
> where is this kwargs coming from

I have tried to answer this question with the code comment added right above that line, and all I know is that kwargs _may_ contain `start` and `stop`. I could not infer any guarantees.

Following the kwargs flow(s) is a challenging part of reading `io/pytables.py`. It's pretty non-obvious.


There are two calls into that function that pass `**kwargs` where it's non-obvious where it's coming from:

https://github.com/pandas-dev/pandas/blob/d47947a79099d650b4bb981a38ce09881ec6f43d/pandas/io/pytables.py#L3888

https://github.com/pandas-dev/pandas/blob/d47947a79099d650b4bb981a38ce09881ec6f43d/pandas/io/pytables.py#L4143

For following just the latter flow I am now looking for `read()` being called and this is where things get tricky...
```
$ grep -nr 'read(' pandas/io/pytables.py | grep -v 'def ' | grep -v validate_read
721:            return s.read(start=_start, stop=_stop,
837:            objs = [t.read(where=_where, columns=columns, start=_start,
1421:        return s.read(**kwargs)
2934:            sdict[c] = s.read()
4241:        s = super().read(columns=columns, **kwargs)
4342:        df = super().read(**kwargs)
4740:        return self.table.table.read(start=self.start, stop=self.stop)
```
From here it looks like there is more kwargs to follow. I know one definite call path that comes along line 837 and there we _know_ that `start` and `stop` are being passed explicitly (that's the code path coming from `read_hdf()`). But I am not sure about other possible code paths.

Relying on the test suite to not break, but also relying on _not_ relying on `start` and `stop` being passed explicitly.

I am now doing this instead:
```
-                      errors=self.errors, **kwargs)
+                      errors=self.errors, start=kwargs.get('start'),
+                      stop=kwargs.get('stop'))
```
So that things still don't break when `start` or `stop` are not in `kwargs`. Sound good?